telegram/files/file.py

```diff
@@ -90,15 +89,15 @@ def __init__(
         self.file_unique_id = str(file_unique_id)
         # Optionals
         self.file_size = file_size
-        self.file_path = file_path
+        self.file_path: Optional[str] = file_path  # Can be either path or url, so must remain str.
         self.bot = bot
         self._credentials: Optional['FileCredentials'] = None
 
         self._id_attrs = (self.file_unique_id,)
 
     def download(
-        self, custom_path: str = None, out: IO = None, timeout: int = None
-    ) -> Union[str, IO]:
+        self, custom_path: Optional[Union[Path, str]] = None, out: IO = None, timeout: int = None
+    ) -> Union[Path, str, IO]:
```

Comment:
please update the docs of the return value accordingly and add a `.. versionchanged`` directive (btw the PR template is there for a reason 😉 )

Reply:
in fact, you can skip the `str`. IISC only `Path` or `IO` is returned …

Reply:
done